projects/Mallard/src/helpers/fetch.ts

```diff
@@ -139,4 +139,47 @@ const fetchFromIssue = <T>(
     )
 }
 
-export { fetchFromIssue, fetchFromApi }
+const fetchFromWeatherApi = async <T>(
+    path: string,
+    {
+        validator = () => true,
+    }: {
+        validator?: ValidatorFn<T>
+    } = {},
+): Promise<T> => {
+    return fetch('http://mobile-weather.guardianapis.com/' + path)
+        .then(res => {
+            if (res.status >= 500) {
+                throw new Error('Failed to fetch') // 500s don't return json
+            }
+            return res.json()
+        })
+        .then(data => {
+            if (data && validator(data)) {
+                return data
+            } else {
+                throw new Error(REQUEST_INVALID_RESPONSE_VALIDATION)
+            }
+        })
+}
+
+const fetchWeather = <T>(
+    endpointPath: string,
+    { validator }: { validator?: ValidatorFn<T> } = {},
+): CachedOrPromise<T> => {
+    return createCachedOrPromise(
+        [
+            null,
```

Comment:
we should cache the weather here, this cache is at runtime so it's about preventing the request from firing more times than needed rather than a long term thing